sklearn_numba_dpex/kmeans/kernels/utils.py

```diff
@@ -58,6 +59,68 @@ def centroid_shifts(
     return centroid_shifts[global_size, work_group_size]
 
 
+@lru_cache
+def make_reduce_centroid_data_kernel(
```

Comment:
this kernel is the fusion of `make_sum_reduction_2d_axis0_kernel` and `make_sum_reduction_3d_axis0_kernel` + some extra instructions to register empty clusters.